src/plugins/intel_cpu/src/nodes/gather.cpp

```diff
@@ -41,8 +42,64 @@ bool Gather::isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std
     return true;
 }
 
+class GatherShapeInfer : public ShapeInferEmptyPads {
+public:
+    GatherShapeInfer(bool isAxisInputConst, int axis, int batchDims, IShapeInfer::port_mask_t port_mask) :
+                     m_isAxisInputConst(isAxisInputConst), m_axis(axis), m_batchDims(batchDims), m_port_mask(port_mask) {}
+    Result infer(const std::vector<std::reference_wrapper<const VectorDims>>& input_shapes,
+                 const std::unordered_map<size_t, MemoryPtr>& data_dependency) override {
+        static constexpr size_t GATHER_DATA = 0, GATHER_INDICES = 1, GATHER_AXIS = 2;
+
+        auto input_shape = input_shapes[GATHER_DATA].get();
+        auto indices_shape = input_shapes[GATHER_INDICES].get();
+
+        if (!m_isAxisInputConst)
+            m_axis = reinterpret_cast<const int32_t *>(data_dependency.at(GATHER_AXIS)->GetPtr())[0];
+
+        if (m_axis < 0)
+            m_axis += input_shape.size();
+        if (m_batchDims < 0)
+            m_batchDims += indices_shape.size();
+
+        VectorDims output_shape;
+        output_shape.reserve(input_shape.size() + indices_shape.size() - m_batchDims - 1);
+        output_shape.insert(output_shape.end(), input_shape.begin(), input_shape.begin() + m_axis);
+        output_shape.insert(output_shape.end(), indices_shape.begin() + m_batchDims, indices_shape.end());
+        output_shape.insert(output_shape.end(), input_shape.begin() + m_axis + 1, input_shape.end());
+
+        return {{output_shape}, ShapeInferStatus::success};
+    }
+    port_mask_t get_port_mask() const override {
+        return m_port_mask;
+    }
+
+private:
+    bool m_isAxisInputConst = false;
+    int m_axis = 0;
+    int m_batchDims = 0;
+    port_mask_t m_port_mask;
+};
+
+class GatherShapeInferFactory : public ShapeInferFactory {
+public:
+    GatherShapeInferFactory(std::shared_ptr<ov::Node> op, IShapeInfer::port_mask_t port_mask) : m_op(op), m_port_mask(port_mask) {}
```

Comment:
```suggestion
    GatherShapeInferFactory(std::shared_ptr<ov::Node> op) : m_op(op), m_port_mask(port_mask) {}
```
There is no need to pass the port mask as a parameter. We can simply return `PortMask(GATHER_AXIS)` in the `get_port_mask()` call.

Reply:
Applied. Thanks Maksim!